pkg/signatures/factory.go

```diff
@@ -120,7 +120,7 @@ func FetchImageSignaturesWithRetries(ctx context.Context, fetcher SignatureFetch
 	var fetchedSignatures []*storage.Signature
 	var err error
 	err = retry.WithRetry(func() error {
-		fetchedSignatures, err = fetchAndAppendSignatures(ctx, fetcher, image, fullImageName, registry, fetchedSignatures)
+		fetchedSignatures, err = fetchAndAppendSignatures(ctx, fetcher, image, fullImageName, registry)
```

Comment:
The change looks good, but this function does not do the `AndAppend` part anymore. Should it be renamed maybe?

Reply:
Yes it should, thanks for the catch.

Reply:
renamed to `fetchUniqueSignatures`.